src/frontend/planner_test/tests/testdata/output/sink.yaml

```diff
@@ -106,15 +108,17 @@
     ├── columns:
     │   ┌── kv_log_store_epoch: bigint
     │   ├── kv_log_store_seq_id: integer
+    │   ├── kv_log_store_vnode: smallint
     │   ├── kv_log_store_row_op: smallint
     │   ├── v1: integer
     │   ├── v2: double precision
     │   ├── v3: character varying
     │   └── v5: numeric
-    ├── primary key: [ $0 ASC, $1 ASC ]
-    ├── value indices: [ 0, 1, 2, 3, 4, 5, 6 ]
-    ├── distribution key: [ 5, 6 ]
-    └── read pk prefix len hint: 2
+    ├── primary key: [ $0 ASC, $1 ASC, $2 ASC ]
+    ├── value indices: [ 0, 1, 2, 3, 4, 5, 6, 7 ]
+    ├── distribution key: [ 6, 7 ]
```

Comment:
Shouldn't the dist key be the vnode column?

Reply:
The dist key is used to compute the vnode from the payload column. Before computing vnode by the dist key, we cannot get vnode from the column. Moreover, if we set the vnode column as the dist key, we will not get the vnode directly from the column, but instead hash on the vnode to get a hashed value, which is different to the vnode.

In batch query, since the `vnode_col_idx` is set, it will read the vnode column directly as the vnode.

Reply:
> The dist key is used to compute the vnode from the payload column. Before computing vnode by the dist key, we cannot get vnode from the column.

Can you elaborate more? I may be missing something, but I found that the backfill state table use vnode column as the dist key as well as vnode column idx. 

https://github.com/risingwavelabs/risingwave/blob/9781d5fe015c42e27cc9edaa60635941207adc88/src/frontend/planner_test/tests/testdata/output/union.yaml#L42

Reply:
The dist key indices will be ignored because when computing vnode in state table and storage table, vnode column idx is perfered than dist key idx if it is set. In this backfill state table, the vnode column idx is set, and therefore the dist key idx is ignored. Since hash over a vnode will not get the original vnode value, if we use dist key (set as vnode idx) to compute the vnode, we will get a different vnode than the original vnode.

Reply:
I see. Does it mean that as long as `vnode_col_idx` is set, `distribution key` is useless and what is in `distribution key` doesn't matter?

Reply:
Discussed offlone: log store writer will use the `distribution key` to calculate the vnode value for serialization so `distribution key` must be the correct one.